lib/fluent/config/literal_parser.rb

```diff
@@ -81,7 +91,13 @@ def scan_double_quoted_string
         string = []
         while true
           if skip(/\"/)
-            return string.join
+            if string.include?(nil)
```

Comment:
If a user writes config like `"TEST#{ENV['A'] || use_nil}"`, the actual value becomes `nil`, not `"TEST"` (I mean, we can use `use_nil` or `use_default` even if there is other values).  Is it intentional?

Reply:
Yes, it's intentional. But I'm not sure it's desired behavior or not. At first I implemented it as `string == [nil]` but replaced it with `string.include?(nil)` for considering some cases like following:

```
somepath "#{ENV['HOME'] || use_nil}/path/to/sub/dir"
```

In this case users might not want replacing it with "/path/to/sub/dir".